packages/client/scripts/default-index.d.ts

```diff
@@ -44,7 +44,11 @@ export declare type PrismaClientExtends<
   >(
     args:
       | ((client: PrismaClientExtends<ExtArgs>) => { $extends: { extArgs: Args } })
-      | { name?: string; result?: R; model?: M; query?: Q; client?: C },
+      | { name?: string }
+      | { result?: R & runtime.Types.Extensions.UserArgs['result'] }
```

Comment:
The split into a union is just for the beauty of the code. The actual fix is to intersect `R` with `runtime.Types.Extensions.UserArgs['result']`, although `R` already extends `runtime.Types.Extensions.UserArgs['result']`. This is why I think it is more something to do with TS than the actual code.